bin/improver-spot-extract

```diff
@@ -245,36 +236,41 @@ def main():
         if not files_to_read:
             raise IOError('No relevant data files found in {} for {}.'.format(
                 args.data_path, diagnostic_filepath))
-        diagnostic_name = diagnostics[diagnostic_key]["diagnostic_name"]
         # Load cubes into an iris.cube.CubeList.
-        diagnostics[diagnostic_key]["data"] = Load('multi_file').process(
-            files_to_read, diagnostic_name)
+        diagnostics[diagnostic_key]["data"] = (
+            Load('multi_file').process(files_to_read))
         # Check if additional diagnostics are needed (e.g. multi-level data).
         # If required, load into the additional_diagnostics dictionary.
         diagnostics[diagnostic_key]["additional_data"] = (
             get_method_prerequisites(
                 diagnostics[diagnostic_key]["interpolation_method"],
-                args.data_path))
+                os.path.dirname(args.data_path)))
 
     resulting_cubes, extrema_cubes = (
         run_spotdata(
             diagnostics, ancillary_data, sites, config_constants,
-            forecast_date=args.forecast_date, forecast_time=args.forecast_time,
-            forecast_length=args.forecast_length,
             use_multiprocessing=args.multiprocess))
 
-    for resulting_cube in resulting_cubes:
-        if resulting_cube:
-            WriteOutput(
-                'as_netcdf', dir_path=args.output_path).process(resulting_cube)
+    filename = os.path.splitext(os.path.basename(all_available_files[0]))[0]
+
+    if resulting_cubes:
+        # Concatenate CubeList into Cube, creating a time DimCoord,
+        # and write out.
+        cube_out, = resulting_cubes.concatenate()
```

Comment:
NB: This won't work when processing multiple diagnostics. Both this and the saving will need to be moved within the diagnostics loop above.